dso/model.py

```diff
@@ -537,6 +546,107 @@ def purge_old_states(
             self.state.remove(state)
 
 
+class FalcoPriority(enum.StrEnum):
+    EMERGENCY = 'Emergency'
+    ALERT = 'Alert'
+    CRITICAL = 'Critical'
+    ERROR = 'Error'
+    WARNING = 'Warning'
+    NOTICE = 'Notice'
+    INFORMATIONAL = 'Informational'
+    DEBUG = 'Debug'
+
+
+@dataclasses.dataclass(frozen=True)
+class FalcoEvent:
+    message: str
+    cluster: str
+    hostname: str
+    time: datetime.datetime
+    rule: str
+    priority: FalcoPriority
+    output: dict[str, typing.Any]
+
+
+@dataclasses.dataclass(frozen=True)
+class ExceptionTemplate:
+    template: str
+
+
+@dataclasses.dataclass(frozen=True)
+class Node:
+    name: str
+    count: int
+
+
+@dataclasses.dataclass(frozen=True)
+class Cluster:
+    name: str
+    nodes: list[Node]
+
+
+@dataclasses.dataclass(frozen=True)
+class FalcoEventGroup(Finding):
+    """
+    FalcoEventGroup represents a group of Falco events that are similar in
+    nature. In almost all cases those are false posities and can be ignored.
+    Falco exceptions can be defined but they can be silenced here.
+    """
+    message: str
+    clusters: list[Cluster]
+    landscape: str
+    project: str
+    rule: str
+    priority: FalcoPriority
+    """Falco priority, one of EMERGENCY, ALERT, CRITICAL, ERROR, WARNING,
+    NOTICE, INFORMATIONAL, DEBUG
+    """
+    first_event: datetime.datetime
+    last_event: datetime.datetime
+    count: int
+    """number of events in this group"""
+    group_hash: str
+    """hash of the group (event fiields and values that form the group),
+    can be reconstructed from a sample event and the fields property."""
+    fields: dict[str, str]
+    """Identical fields that form the group"""
+    events: list[FalcoEvent]
+    """list of events in this group (possibly truncated)."""
+    exception: ExceptionTemplate
+    """exception template for this group"""
+
+    @property
+    def key(self) -> str:
+        return self.group_hash
+
+
+@dataclasses.dataclass(frozen=True)
+class FalcoDebugEventGroup(Finding):
+    """
+    Group of events that - most likely - are a result of a single debug
+    session. It might however also be an indication of an attack. These
+    events must be reviewed and ideally be linked to some legal activity.
+    """
+    count: int
+    cluster: str
+    hostname: str
+    project: str
+    landscape: str
+    group_hash: str
+    """reproducible group hash to avoid double reporting should the reporting
```

Comment:
nit: as attribute-specific docstrings are not available at runtime ([ref](https://bugs.python.org/issue38401)), it might be nice to document attributes in type's docstr (the first one)